azure-client-runtime/src/main/java/com/microsoft/azure/AzureProxy.java

```diff
@@ -140,72 +141,108 @@ protected Object handleAsyncHttpResponse(final HttpRequest httpRequest, Single<H
+                            pollStrategy = new CompletedPollStrategy(AzureProxy.this, originalHttpResponse);
+                        }
+
+                        if (pollStrategy != null) {
+                            result = Single.just(pollStrategy);
                         }
 
                         if (result == null) {
-                            final HttpResponse bufferedOriginalHttpRespose = originalHttpResponse.buffer();
-                            result = bufferedOriginalHttpRespose.bodyAsStringAsync()
-                                .map(new Func1<String, PollStrategy>() {
-                                    @Override
-                                    public PollStrategy call(String originalHttpResponseBody) {
-                                        PollStrategy result = null;
-
-                                        try {
-                                            final ResourceWithProvisioningState resource = serializer.deserialize(originalHttpResponseBody, ResourceWithProvisioningState.class);
-                                            if (resource != null && resource.properties() != null && !ProvisioningState.isCompleted(resource.properties().provisioningState())) {
-                                                result = new ProvisioningStatePollStrategy(AzureProxy.this, originalHttpRequest, resource.properties().provisioningState(), delayInMilliseconds);
-                                            }
-                                        } catch (IOException e) {
-                                        }
-
-                                        if (result == null) {
-                                            result = new CompletedPollStrategy(AzureProxy.this, bufferedOriginalHttpRespose);
-                                        }
-
-                                        return result;
-                                    }
-                                });
+                            throw new CloudException("Could not determine a long running operation polling strategy.", originalHttpResponse);
                         }
 
                         return result;
                     }
                 });
     }
+
+    private Single<PollStrategy> createProvisioningStateOrCompletedPollStrategy(final HttpRequest httpRequest, HttpResponse httpResponse, SwaggerMethodParser methodParser, final long delayInMilliseconds) {
+        Single<PollStrategy> result;
+
+        final String httpRequestMethod = httpRequest.httpMethod();
+        if (httpRequestMethod.equalsIgnoreCase("DELETE") ||
+                httpRequestMethod.equalsIgnoreCase("GET") ||
+                httpRequestMethod.equalsIgnoreCase("HEAD") ||
+                !methodParser.expectsResponseBody()) {
+            result = Single.<PollStrategy>just(new CompletedPollStrategy(AzureProxy.this, httpResponse));
```

Comment:
consider checking content-type: application/json here

Reply:
This way we won't have to buffer octet streams and attempt to parse them as JSON before we give them over

Reply:
My only fear is that autorest.testserver isn't giving a response Content-Type back--but we'll be able to work around it, maybe even with a requestpolicy that adds it in the worst case